pkg/apiserver/experiment/experiment.go

```diff
@@ -857,74 +868,73 @@ func (s *Service) deleteExperiment(c *gin.Context) {
+		if chaosKind, ok = v1alpha1.AllKinds()[kind]; !ok {
+			joinErrString(&errStr, fmt.Sprintf("delete experiment uid (%s) error, because kind (%s) is not supported", uid, kind))
+			continue
 		}
-		return
-	}
-
-	if force == "true" {
-		if chaosMeta, ok = chaosKind.Chaos.(metav1.Object); !ok {
-			c.Status(http.StatusInternalServerError)
-			_ = c.Error(utils.ErrInternalServer.WrapWithNoMessage(fmt.Errorf("failed to get chaos meta information")))
-			return
+		if err := kubeCli.Get(ctx, chaosKey, chaosKind.Chaos); err != nil {
+			if apierrors.IsNotFound(err) {
+				joinErrString(&errStr, fmt.Sprintf("delete experiment uid (%s) error, because the chaos is not found", uid))
+			} else {
+				joinErrString(&errStr, fmt.Sprintf("delete experiment uid (%s) error, because %s", uid, err.Error()))
+			}
+			continue
 		}
 
-		annotations := chaosMeta.GetAnnotations()
-		if annotations == nil {
-			annotations = make(map[string]string)
-		}
-		annotations[common.AnnotationCleanFinalizer] = common.AnnotationCleanFinalizerForced
-		chaosMeta.SetAnnotations(annotations)
-		if err := kubeCli.Update(context.Background(), chaosKind.Chaos); err != nil {
-			c.Status(http.StatusInternalServerError)
-			_ = c.Error(utils.ErrInternalServer.WrapWithNoMessage(fmt.Errorf("forced deletion of chaos failed, because update chaos annotation error")))
-			return
+		if force == "true" {
+			if chaosMeta, ok = chaosKind.Chaos.(metav1.Object); !ok {
+				joinErrString(&errStr, fmt.Sprintf("delete experiment uid (%s) error, because failed to get chaos meta information", uid))
+				continue
+			}
+
+			annotations := chaosMeta.GetAnnotations()
+			if annotations == nil {
+				annotations = make(map[string]string)
+			}
+			annotations[common.AnnotationCleanFinalizer] = common.AnnotationCleanFinalizerForced
+			chaosMeta.SetAnnotations(annotations)
+			if err := kubeCli.Update(context.Background(), chaosKind.Chaos); err != nil {
+				joinErrString(&errStr, fmt.Sprintf("forced delete experiment uid (%s) error, because update chaos annotation error", uid))
+				continue
+			}
 		}
-	}
 
-	if err := kubeCli.Delete(ctx, chaosKind.Chaos, &client.DeleteOptions{}); err != nil {
-		if apierrors.IsNotFound(err) {
-			c.Status(http.StatusNotFound)
-			_ = c.Error(utils.ErrNotFound.NewWithNoMessage())
-		} else {
-			c.Status(http.StatusInternalServerError)
-			_ = c.Error(utils.ErrInternalServer.WrapWithNoMessage(err))
+		if err := kubeCli.Delete(ctx, chaosKind.Chaos, &client.DeleteOptions{}); err != nil {
+			if apierrors.IsNotFound(err) {
+				joinErrString(&errStr, fmt.Sprintf("forced delete experiment uid (%s) error, because the chaos is not found", uid))
+			} else {
+				joinErrString(&errStr, fmt.Sprintf("delete experiment uid (%s) error, because %s", uid, err.Error()))
+			}
+			continue
 		}
-		return
 	}
-
-	c.JSON(http.StatusOK, StatusResponse{Status: "success"})
+	if errStr != "" {
+		c.Status(http.StatusInternalServerError)
```

Comment:
That is making it impossible to return a `404 not found`, along with loss of error type information.

Reply:
Yes, but it is possible that err has different types. There is no better idea to distinguish different error types.